apps/backend/app/api/routes/user_data.py

```
@@ -0,0 +1,354 @@
+        # Read the file content
+        file_content = response["Body"].read()
+
+        # Determine file type and read accordingly
+        if user_data.filename.endswith(".csv"):
+            df = pd.read_csv(io.BytesIO(file_content))
+        elif user_data.filename.endswith(".xlsx"):
+            df = pd.read_excel(io.BytesIO(file_content))
+        elif user_data.filename.endswith(".txt"):
+            df = pd.read_csv(io.BytesIO(file_content), sep="\t")
+        else:
+            raise HTTPException(status_code=400, detail="Unsupported file type")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)